cronutils/error_handler.py

```diff
@@ -96,31 +96,61 @@ def raise_errors(self):
             stderr.write(output)
             stderr.write("\n\n")
             raise BundledError()
+
+    @staticmethod
+    def format_traceback_key(exec_value, traceback):
+        return repr(exec_value) + "\n" + str().join(format_tb(traceback))
         
 
 class ErrorSentry(ErrorHandler):
-    
-    def __init__(self, sentry_dsn, descriptor=None, data_limit=100, sentry_client_kwargs=None):
-        
+    """
+    Like an ErrorHandler, but reports errors to a Sentry DSN.
+    Note that sentry_report_limit is a per-stacktrace limit.
+    """
+
+    def __init__(self, sentry_dsn, descriptor=None, data_limit=100, sentry_client_kwargs=None,
+            sentry_report_limit=0
+    ):
         if sentry_client_kwargs:
             self.sentry_client = SentryClient(dsn=sentry_dsn, **sentry_client_kwargs)
         else:
             self.sentry_client = SentryClient(dsn=sentry_dsn)
             
         super(ErrorSentry, self).__init__(descriptor=descriptor, data_limit=data_limit)
+        self.sentry_report_limit = sentry_report_limit
 
     def __exit__(self, exec_type, exec_value, traceback):
         ret = super(ErrorSentry, self).__exit__(exec_type, exec_value, traceback)
-        
+
         if ret and isinstance(exec_value, Exception):
-            self.sentry_client.captureException(exc_info=True)
+            # Identify the whether the report limit has been hit, handle zero/negative value behavior.
+            # (stacktrace as key is guaranteed in the above super call.)
+            traceback_key = self.format_traceback_key(exec_value, traceback)
+            report_limit_not_exceeded = (
+                    self.sentry_report_limit < 1 or
```

Comment:
I think it would be clearer to use a report limit of `None` for the case where we are not doing report limiting.